color_helper.py

```diff
@@ -1001,15 +1001,30 @@ def is_enabled(self, mode, palette_name=None, color=None, auto=False):
         )
 
 
-class ColorHelperFileIndexCommand(sublime_plugin.TextCommand):
-    """Color Helper file index command."""
+class ColorHelperFileIndexCommand(sublime_plugin.WindowCommand):
+    """Forces Color Helper to index the colors in the current file.
 
-    def run(self, edit):
+    If `scan_scopes` argument is provided, instead of using the normal
+    configuration it will look for colors inside the given scopes.
+    """
+
+    def run(self, scan_scopes=None, allowed_colors=["all"], use_hex_argb=False):
         """Run the command."""
-        rules = util.get_rules(self.view)
-        if rules and util.get_scope(self.view, rules, skip_sel_check=True):
+        view = self.window.active_view()
+        if scan_scopes:
+            if "all" in allowed_colors:
```

Comment:
So you are manually handling `"all"`.  Do you need to manually handle others like `"css3"`?  Do we need a function to resolve allowed colors in this situation?

Reply:
extracted the logic in `util.expand_color`, handling of `"css3"` and `"css4"` is now correctly done.